### PR TITLE
[VEGA-4071] fix(components): add handler distribution cannot be restored error

### PR DESCRIPTION
**Pull request by troynin**
[Jira issue: VEGA-4071](https://jira.vegaspace.tk/browse/VEGA-4071)

## Problem statement

[FE] [BE] Обработка ошибок восстановления распределений
[https://jira.vegaspace.tk/browse/VEGA-4071](https://jira.vegaspace.tk/browse/VEGA-4071)

## Solution details

Добавил вывод ошибки

stand: http://it-4071-lanit-vega-builder.code013.org/

e2e: https://git.konakov.tv/vega2/vega-scenario-tests/-/merge_requests/76

## Type

- [ ] Feature
- [x] Bug Fix
- [ ] Refactoring
- [x] Test
- [ ] Other

## Associated issues

_{Jira issues or PRs related to this one}_

## Quality control

- [x] PR: Based on a correct branch
- [x] PR: Head branch is rebased on the base branch
- [ ] PR: Assignee chosen and necessary labels are set
- [ ] PR: Current form is filled out (where it makes sense)
- [ ] PR: Diff checked for irrelevant changes
- [x] PR: Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specs
- [x] PR: Checked for spelling, grammar and typos
- [ ] JS: There's no errors/warnings in the browser dev console
- [ ] Layout: Tested with various content amounts
- [ ] Docs: All the important changes and features are described
- [ ] Tests: Configuration files are checked at test repositories
- [x] Tests: New unit tests developed
- [x] Tests: New E2E tests developed
- [x] Tests: Existing unit tests passed successfully
image
- [x] Tests: Existing E2E tests passed successfully
image
- [x] Tests: Manually tested on personal instance

## Notable statements
- [ ] Affects configuration files
- [ ] Brings new packages or updates existing
- [ ] Opened follow-up tasks to resolve

### CODE REVIEW

File: src/components/DistributionSettings/DistributionSettingsForm/DistributionSettingsForm.tsx [hunk 1]
```diff
@@ -163,6 +170,9 @@ export const DistributionSettingsForm: React.FC<IProps> = ({
           ]?.map((field, index, fields) =>
             renderFormField(field, index, fields),
           )}
+          {commonErrors.map((error) => (
+            <div className={cnForm('ErrorMessage')}>{error.message}</div>
```
Comment (lipgen) [inline src/components/DistributionSettings/DistributionSettingsForm/DistributionSettingsForm.tsx:174]: Думаю, нужно написать тест, чтобы проверить этот функционал

File: src/components/DistributionSettings/DistributionSettingsForm/DistributionSettingsForm.test.tsx [hunk 2]
```diff
@@ -173,4 +174,35 @@ describe('DistributionSettingsForm', () => {
...
+        __typename: 'CommonError',
+      },
+    ];
+    const { baseElement } = renderComponent({
+      onChange: () => {},
+      formData: initialFormData,
+      errors,
+      chart: null,
+    });
+    console.log(baseElement);
```
Comment (lipgen) [inline src/components/DistributionSettings/DistributionSettingsForm/DistributionSettingsForm.test.tsx:202]: нужно убрать

File: src/components/DistributionSettings/DistributionSettingsForm/DistributionSettingsForm.test.tsx [hunk 3]
```diff
@@ -173,4 +174,35 @@ describe('DistributionSettingsForm', () => {
...
+    const errorBlock = baseElement.querySelectorAll(
+      '.DistributionSettingsForm__ErrorMessage',
+    );
+    expect(errorBlock).toHaveLength(1);
```
Comment (lipgen) [inline src/components/DistributionSettings/DistributionSettingsForm/DistributionSettingsForm.test.tsx:206]: я бы еще добавил проверку, что отображается тот текст который нужно

File: src/components/DistributionSettings/DistributionSettingsForm/DistributionSettingsForm.test.tsx (hunk 2 above)
Comment (troynin) [reply src/components/DistributionSettings/DistributionSettingsForm/DistributionSettingsForm.test.tsx]: сделал

File: src/components/DistributionSettings/DistributionSettingsForm/DistributionSettingsForm.test.tsx (hunk 3 above)
Comment (troynin) [reply src/components/DistributionSettings/DistributionSettingsForm/DistributionSettingsForm.test.tsx:205]: сделал

**Review by lipgen — APPROVED** [review]

**Pull request merged**